homeassistant/components/viam/config_flow.py

```diff
@@ -0,0 +1,200 @@
+"""Config flow for viam integration."""
+from __future__ import annotations
+
+import logging
+from typing import Any
+
+from viam.app.viam_client import ViamClient
+from viam.rpc.dial import Credentials, DialOptions
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResult
+from homeassistant.exceptions import HomeAssistantError
+from homeassistant.helpers.selector import selector
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_AUTH_USER_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("credential_type"): selector(
+            {
+                "select": {
+                    "options": [
+                        {"value": "api-key", "label": "Org API Key"},
+                        {
+                            "value": "robot-location-secret",
+                            "label": "Robot Location Secret",
+                        },
+                    ],
+                    "translation_key": "credential_type",
+                }
+            }
+        )
+    }
+)
+STEP_AUTH_ROBOT_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("address"): str,
+        vol.Required("secret"): str,
+    }
+)
+STEP_AUTH_ORG_DATA_SCHEMA = vol.Schema(
+    {
+        vol.Required("api_id"): str,
+        vol.Required("api_key"): str,
```

Comment:
```suggestion
        vol.Required(CONF_API_KEY): str,
```

Check if for the other keys also constants exists